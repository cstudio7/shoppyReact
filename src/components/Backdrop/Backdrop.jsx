import React from 'react';
import PropTypes from 'prop-types';
import './Backdrop.scss';

const backdrop = props => <div className="backdrop" onClick={props.click} role="presentation" />;

backdrop.propTypes = {
  click: PropTypes.func.isRequired
};

export default backdrop;
