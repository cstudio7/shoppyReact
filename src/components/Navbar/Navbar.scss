.mainContainer {
  height: 100%;
}
.pageContent {
  margin-top: 64px;
}

@media (min-width: 769px) {
  .pageContent {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
  }
}
