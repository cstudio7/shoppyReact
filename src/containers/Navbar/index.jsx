import React, { Component } from 'react';
import PropTypes from 'prop-types';
import NavbarComponent from '../../components/Navbar';

export default class Navbar extends Component {
  render() {
    return <NavbarComponent userRole="admin" displayPage={this.props.displayPage} />;
  }
}
Navbar.propTypes = {
  displayPage: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]).isRequired
};
