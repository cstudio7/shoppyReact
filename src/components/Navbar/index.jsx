import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Toolbar from '../Toolbar/Toolbar';
import SideDrawer from '../SideDrawer/SideDrawer';
import Backdrop from '../Backdrop/Backdrop';
import './Navbar.scss';

class Navbar extends Component {
  state = {
    sideDrawerOpen: false
  };

  drawerToggleClickHandler = () => {
    this.setState(prevState => ({ sideDrawerOpen: !prevState.sideDrawerOpen }));
  };

  backdropClickHandler = () => {
    this.setState({ sideDrawerOpen: false });
  };

  render() {
    let backdrop;

    if (this.state.sideDrawerOpen) {
      backdrop = <Backdrop click={this.backdropClickHandler} />;
    }
    return (
      <div className="mainContainer">
        <Toolbar drawerClickHandler={this.drawerToggleClickHandler} />
        <SideDrawer show={this.state.sideDrawerOpen} />
        {backdrop}
        <main className="pageContent">{this.props.displayPage}</main>
      </div>
    );
  }
}

Navbar.propTypes = {
  displayPage: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]).isRequired
};

export default Navbar;
