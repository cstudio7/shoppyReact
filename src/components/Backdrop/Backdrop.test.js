import React from 'react';
import { shallow } from 'enzyme';
import Backdrop from './Backdrop';
import Navbar from '../Navbar';

describe('Test the Backdrop Component', () => {
  it('should match snapshot', () => {
    const click = jest.fn();
    const wrapper = shallow(<Backdrop click={click} />);
    expect(wrapper).toMatchSnapshot();
  });
});
