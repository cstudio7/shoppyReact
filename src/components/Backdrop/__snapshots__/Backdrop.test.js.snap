// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Backdrop Component should match snapshot 1`] = `
<div
  className="backdrop"
  onClick={[MockFunction]}
  role="presentation"
/>
`;
